framework/include/utils/InputParametersChecksUtils.h

```diff
@@ -353,9 +358,43 @@ InputParametersChecksUtils<C>::checkVectorParamsNoOverlap(
 
     for (const auto & value : forwardGetParam<std::vector<T>>(param))
       if (!unique_params.insert(value).second)
+      {
+        auto copy_params = param_vec;
+        copy_params.erase(std::find(copy_params.begin(), copy_params.end(), param));
```

Comment:
What is the purpose of these two lines? To me, it looks like a local copy is made, a value is erased, and then the copy is never used.

Reply:
it's to avoid saying
```
"Item '" xyz "' specified in vector parameter '" + parameter1 +
                            "' is also present in one or more of the two-D vector parameters '" +
                            "parameter1 parameter2 parameter3" +
                            "' in the inner vector of the same index, which is not allowed.");
```
which isnt clear to the user since parameter1's name shows up twice

Reply:
I think @pbehne is suggesting you should have `stringify(copy_params)` instead of `stringify(param_vec)`

Reply:
ah yes that's what I intended. good catch